internal/provider/resource_machine.go

```diff
@@ -0,0 +1,169 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/juju/terraform-provider-juju/internal/juju"
+)
+
+func resourceMachine() *schema.Resource {
+	return &schema.Resource{
+		Description: "A resource that represents a Juju machine deployment. Refer to the juju add-machine CLI command for more information and limitations.",
+
+		CreateContext: resourceMachineCreate,
+		ReadContext:   resourceMachineRead,
+		DeleteContext: resourceMachineDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "A name for the machine resource in Terraform.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+			},
+			"model": {
+				Description: "The Juju model in which to add a new machine.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"constraints": {
+				Description: "Machine constraints that overwrite those available from 'juju get-model-constraints' and provider's defaults.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				ForceNew:    true,
+			},
+			"disks": {
+				Description: "Storage constraints for disks to attach to the machine(s).",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				ForceNew:    true,
+			},
+			"series": {
+				Description: "The operating system series to install on the new machine(s).",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"machine_id": {
+				Description: "The id of the machine Juju creates.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
```

Comment:
I think required and optional should be `false` to make this attribute read only.

Reply:
That works for me. I will make that cahnge.

Reply:
I made the attribute read only as you requested. Thank you again!